### PR TITLE
Colorbar levels as list/array in gridmap & small fixes

### PR DESCRIPTION
**Pull request by vindelico**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
  - This PR fixes small bugs
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] CHANGES.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

*  make deepcopy of input dict to avoid changing it; (util.py)
*  keep reference to a Dataset if data is passed as DataArray (plot.py)
* allow passing predefined colorbar levels as list/np.ndarray.
* rename extend->extent (plot.py)

### Does this PR introduce a breaking change?

No.

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@vindelico if you want to not run into formatting issues, run:
```
$ mamba install pre-commit
$ pre-commit install
```

File: CHANGES.rst [hunk 1]
```diff
@@ -19,6 +19,7 @@ Contributors to this version: Sarah-Claude Bourdeau-Goulet (:user:`Sarahclaude`)
 
 New features and enhancements
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* Use list or ndarray as levels for colorbar in gridmap and small bug fixes (:pull:`???`).
```
Comment (Zeitsperre) [inline CHANGES.rst:22]:
```suggestion
* Use list or ndarray as levels for colorbar in gridmap and small bug fixes (:pull:`176`).
```

Comment (vindelico) [reply CHANGES.rst:22]: Right, thx!

File: figanos/matplotlib/plot.py [hunk 2]
```diff
@@ -682,15 +683,18 @@ def gridmap(
...
-            cmap,
-            np.nanmin(plot_data.values),
-            np.nanmax(plot_data.values),
-            levels=levels,
-            divergent=divergent,
-            linspace_out=True,
-        )
+    if levels is not None:
+        if isinstance(levels, Iterable):
+            lin = levels
```
Comment (juliettelavoie) [inline figanos/matplotlib/plot.py:688]: fyi, it was already possible to pass a list through `plot_kw={'levels':[290,294,298,302]}`, but nice to access it also directly here.

I am wondering if we should change to example in the notebook for this instead?

File: figanos/matplotlib/plot.py [hunk 3]
```diff
@@ -1543,9 +1547,10 @@ def scattermap(
         else:
             raise ValueError("If `data` is a dict, it must be of length 1.")
 
-    # select data to plot
+    # select data to plot and its xr.Dataset
     if isinstance(data, xr.DataArray):
         plot_data = data
+        data = xr.Dataset({plot_data.name: plot_data})
```
Comment (juliettelavoie) [inline figanos/matplotlib/plot.py:1553]: why does it need to be a dataset ? 
what if it doesn't have a name ?

File: figanos/matplotlib/plot.py [hunk 4]
```diff
@@ -749,8 +753,8 @@ def gridmap(
         im = plot_data.plot.contourf(**plot_kw)
 
     if ax:
-        if extend:
```
Comment (juliettelavoie) [inline figanos/matplotlib/plot.py:752]: good catch!

File: figanos/matplotlib/plot.py (hunk 3 above)
Comment (vindelico) [reply figanos/matplotlib/plot.py:1553]: TBH, it's been a while that I added that so don't quite remember 🤔 ...

It was something missing on the plot if the dataset was not there.

Comment (juliettelavoie) [reply figanos/matplotlib/plot.py:1553]: haha ok!

**Review by juliettelavoie — APPROVED** [review]

File: figanos/matplotlib/plot.py (hunk 2 above)
Comment (sarahclaude) [reply figanos/matplotlib/plot.py:688]: Would be nice to see the option!

**Review by sarahclaude — APPROVED** [review]
Looks good, thanks Marco!

File: CHANGES.rst [hunk 5]
```diff
@@ -19,6 +19,7 @@ Contributors to this version: Sarah-Claude Bourdeau-Goulet (:user:`Sarahclaude`)
 
 New features and enhancements
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* Use list or ndarray as levels for colorbar in gridmap and small bug fixes (:pull:`176`).
```
Comment (Zeitsperre) [inline CHANGES.rst:22]:
```suggestion
* Use list or ndarray as levels for colorbar in gridmap and small bug fixes (:pull:`176`).
```

This should be under v4.0

**Review by Zeitsperre — CHANGES_REQUESTED** [review]

**Review by Zeitsperre — APPROVED** [review]

**Pull request merged**